set8/57/strings/swap.cc

```diff
@@ -0,0 +1,16 @@
+#include "strings.ih"
+
+void Strings::swap(Strings &other)
+{
+    string **tmp = d_data;
+    d_data = other.d_data;
+    other.d_data = tmp;
+
+    size_t size = d_size;
+    d_size = other.d_size;
+    other.d_size = size;
+
+    size_t tmpCapacity = d_capacity;
+    d_capacity = other.d_capacity;
+    other.d_capacity = tmpCapacity;
+}
```

Comment:
```suggestion
void Strings::swap(Strings &other)
{
    std::swap(d_data, other.d_data);
    std::swap(d_size, other.d_size);
    std::swap(d_capacity, other.d_capacity);
}
```

You can use `std::swap` here to make this a bit cleaner.